### PR TITLE
#158 upgraded gradle version to 7.6.4 and changes are comaptible to 8.x ve…

### PR DESCRIPTION
**Pull request by shanki2502**
As part of this PR below changes are done.
1. upgraded plugin versions
2. upgraded library versions
3. upgraded gradle version to 7.6.4
4. updated use of outputDir to classesDirectory.get().asFile

### CODE REVIEW

**Review by Alex-Vol — APPROVED** [review]
The change looks good to me.

Thank you for the PR.

**Comment by Alex-Vol** [comment]
It seems that develocity plugin does not have publishAlways() method. Can you correct the build issue and update the PR.

**Review by Alex-Vol — COMMENTED** [review]
I think publishAlways() should be replaced by this

```
publishing {
  onlyIf {
    true
  }
}
```

**Comment by shanki2502** [comment]
I have updated the PR, thank you for your quick response.

**Comment by shanki2502** [comment]
@Alex-Vol  could you please approve it again, so build can trigger.

File: build.gradle [hunk 1]
```diff
@@ -1,38 +1,33 @@
...
-    id 'org.ajoberstar.reckon' version '0.12.0'
-    id 'org.ajoberstar.git-publish' version '2.1.3'
+    id 'org.ajoberstar.reckon' version '0.18.3'
+    id 'org.ajoberstar.git-publish' version '4.2.2'
 }
 
 group = 'com.bmuschko'
 
 gradlePlugin {
-    plugins {
```
Comment (marek-parfianowicz) [inline build.gradle:15]: Why this change?

File: src/main/groovy/com/bmuschko/gradle/clover/CloverSourceSetUtils.groovy [hunk 2]
```diff
@@ -28,7 +28,7 @@ final class CloverSourceSetUtils {
     private CloverSourceSetUtils() {}
 
     static List<File> getValidSourceDirs(Collection<CloverSourceSet> sourceSets) {
-        def srcDirs = []
+        def List<File> srcDirs = []
```
Comment (marek-parfianowicz) [inline src/main/groovy/com/bmuschko/gradle/clover/CloverSourceSetUtils.groovy:31]: Either `def` or `List<File>`

File: src/main/groovy/com/bmuschko/gradle/clover/CloverPlugin.groovy [hunk 3]
```diff
@@ -331,12 +331,12 @@ class CloverPlugin implements Plugin<Project> {
 
         @CompileDynamic
         private boolean isJavaSourceSetOf(SourceSet sourceSet, Test testTask) {
-            return testTask.ext.originalClasspath.contains(sourceSet.java.outputDir) && !testTask.originalTestClassesDir.contains(sourceSet.java.outputDir)
+            return testTask.ext.originalClasspath.contains(sourceSet.java.classesDirectory.get().asFile) && !testTask.originalTestClassesDir.contains(sourceSet.java.classesDirectory.get().asFile)
```
Comment (marek-parfianowicz) [inline src/main/groovy/com/bmuschko/gradle/clover/CloverPlugin.groovy:334]: Where does this `classesDirectory` property come from? Gradle's standard source set has `classesDirs`, which is a collection of files, not just one file.

File: gradle/wrapper/gradle-wrapper.jar
Comment (marek-parfianowicz) [inline gradle/wrapper/gradle-wrapper.jar:1]: This is a binary file. Origin is unknown. Content must be validated if it's truly a gradle-wrapper.jar, with no viruses.

Comment (Alex-Vol) [reply gradle/wrapper/gradle-wrapper.jar:1]: I agree. I will be doing a verification when I get back from vacation after the 20th of September.

File: src/main/groovy/com/bmuschko/gradle/clover/CloverPlugin.groovy (hunk 3 above)
Comment (Alex-Vol) [reply src/main/groovy/com/bmuschko/gradle/clover/CloverPlugin.groovy:334]: SourceDirectorySet is where this comes from

https://docs.gradle.org/current/dsl/org.gradle.api.file.SourceDirectorySet.html#org.gradle.api.file.SourceDirectorySet:classesDirectory

File: build.gradle (hunk 1 above)
Comment (shanki2502) [reply build.gradle:15]: It was giving issue with earlier configuration, so just fixed the issues.

File: src/main/groovy/com/bmuschko/gradle/clover/CloverSourceSetUtils.groovy (hunk 2 above)
Comment (shanki2502) [reply src/main/groovy/com/bmuschko/gradle/clover/CloverSourceSetUtils.groovy:31]: yes, def can be removed.

**Comment by shanki2502** [comment]
@Alex-Vol @marek-parfianowicz  I have updated PR, kindly approve. I will be going on vacation from tomorrow, will update this PR once I am back if there is any comment left.

**Comment by shanki2502** [comment]
linux build is in queue for more than 40 mins

**Comment by Alex-Vol** [comment]
> mins

I may have to update the GitHub action to fix this. The build has not executed for more than a year so some updates may be required

**Comment by shanki2502** [comment]
@Alex-Vol should I change the workflow file to use 22.04.4 or will you update it?

**Comment by Alex-Vol** [comment]
> @Alex-Vol should I change the workflow file to use 22.04.4 or will you update it?

If you can change it l can try running it. I am on vacation till September 20th so can't do it before then

File: gradle/wrapper/gradle-wrapper.jar
Comment (Alex-Vol) [reply gradle/wrapper/gradle-wrapper.jar:1]: If you reset this file to the one in the master branch it will still work. Just the jar file, then I will merge it. Otherwise I will need to verify its validity after I return.

**Comment by Alex-Vol** [comment]
Reset the gradle-wrapper.jar to the master branch and update the PR. This is the only blocking charge. If necessary I can update the JAR myself when I can after I get back from vacation.

Gradle wrapper JAR doesn't change much between releases so I believe it will just work

**Review by marek-parfianowicz — APPROVED** [review]

**Comment by shanki2502** [comment]
@Alex-Vol can you approve and merge this PR?

File: gradle/wrapper/gradle-wrapper.jar
Comment (Alex-Vol) [inline gradle/wrapper/gradle-wrapper.jar:1]: Can you reset this file to the original?

I will approve and merge after it is no longer changed.

**Comment by Alex-Vol** [comment]
> @Alex-Vol can you approve and merge this PR?

If you reset the jar fike I will be able to merge it.

File: gradle/wrapper/gradle-wrapper.jar
Comment (shanki2502) [reply gradle/wrapper/gradle-wrapper.jar:1]: done

**Comment by shanki2502** [comment]
> > @Alex-Vol can you approve and merge this PR?
> 
> If you reset the jar fike I will be able to merge it.

done, could you please approve and merge?

**Comment by Alex-Vol** [comment]
Made changes to master for the upload v2 issue and squash merged

**Comment by Alex-Vol** [comment]
I am away for an extended time and I am not sure when I can perform the new
release ceremony. I will attempt it remotely although it is a bit complex.
I did not bring a laptop to perform work like this with me.
